code/game/objects/structures/low_wall.dm

```diff
@@ -0,0 +1,484 @@
+			deferred_diagonals |= T_dir
+
+		if(propagate)
+			spawn(0)
+				T.update_connections()
+				T.update_icon()
+
+	//Last chance to connect
+	//Now we will dump cpnnected_cardinals list into a bitfield to make the next section simpler
+	var/wall_dirs_bitfield = 0
+	for (var/d in connected_cardinals)
+		wall_dirs_bitfield |= d
+
+	for (var/d in deferred_diagonals)
+	/*
+		Deferred diagonals list now contains the directions of all fullsize walls which are:
+		1. Within 1 tile of this wall
+		2. At a diagonal to this wall
+		3. Not at a cardinal to any adjacent low wall
+
+		There is still a possibility to have a connection to these walls under condition 2c,
+		That is, if it is adjacent to two other fullsize walls which are already confirmed connections
+
+	*/
+		//Since we have everything in a bitfield, we can compare the diagonals against it.
+		//If both of its cardinals are here, the diagonal will be too
+		if ((wall_dirs_bitfield & d) == d)
+			if (debug)	world << "Connected to a diagonal wall, [direction_to_text(d)] [d], Bitfield [wall_dirs_bitfield]"
+			wall_dirs |= (d)
+			connection_dirs |= d
+
+	connections = dirs_to_corner_states(connection_dirs)
+	wall_connections = dirs_to_corner_states(wall_dirs)
+
+
+	/*
+		Extra last bonus chance to connect.
+		Here we will "upgrade" the wall connections in a few specific cases where the overall connections are better
+	*/
+	for (var/i = 1;i<=4;i++)
+		var/a = text2num(wall_connections[i])
+		var/b = text2num(connections[i])
+
+		//First of all, if we have a single vertical high wall connection, but a cross of total connections
+		//Then we will upgrade the high wall connections to a cross too. this prevents some bugginess
+		if ((((i in list(CORNER_NORTHWEST, CORNER_SOUTHEAST)) && a == CORNER_CLOCKWISE) \
+		|| ((i in list(CORNER_NORTHEAST, CORNER_SOUTHWEST)) && a == CORNER_COUNTERCLOCKWISE)) \
+		&& b in list(5,7))
+			//What a mess, all that determines whether a corner connects to only vertical.
+			//If its in the northwest or southeast corner, and its only connection is clockwise, then that connection is either up or down
+			//Ditto with the other check
+			//The last one is checking if b == a cross or block connection
+			wall_connections[i] = connections[i]
+
+		//Secondly, if we have a cross of two high walls, but a block of total connections,
+		//We upgrade wall connections to a block too.
+		//This basically means treating a diagonal low wall like a high wall
+		else if (a == 5 && b == 7)
+			wall_connections[i] = connections[i]
+
+
+#undef CORNER_NORTHWEST
+#undef CORNER_SOUTHEAST
+#undef CORNER_NORTHEAST
+#undef CORNER_SOUTHWEST
+
+#undef CORNER_NONE
+#undef CORNER_COUNTERCLOCKWISE
+#undef CORNER_DIAGONAL
+#undef CORNER_CLOCKWISE
+
+
+
+
+//Attack handling, damage, deconstructing, etc.
+//Mostly copied and modified from normal walls
+/obj/structure/low_wall/attackby(obj/item/I, mob/user, var/params)
+
+	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+	if (!user)
+		return
```

Comment:
Shouldn't we check that before `user.setClickCooldown`?
All in all, that check seems redundant.